src/cell/mod.rs

```diff
@@ -0,0 +1,31 @@
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub struct Cell {
+    pub x: i32,
+    pub y: i32,
+}
+
+impl Cell {
+    pub fn new(x: i32, y: i32) -> Self {
+        Cell { x, y }
+    }
+
+    // Returns the positions of the 8 neighbours of a cell
+    pub fn neighbour_positions(&self) -> Vec<Cell> {
+        let mut positions = Vec::with_capacity(8);
+
+        for dx in -1..=1 {
+            for dy in -1..=1 {
+                if dx == 0 && dy == 0 {
+                    continue;
+                }
+
+                positions.push(Cell {
+                    x: self.x + dx,
+                    y: self.y + dy,
+                });
+            }
+        }
+
+        positions
+    }
```

Comment:
It might be a good idea for this function to include boundary checks. I'm not sure how you'll handle the upper boundaries, but for now, it should at least prevent `-1` values for both `x` and `y`.

Reply:
Yeah, I think restricting to `u32` should be enough. I haven't figured out how I'll use the UI yet, but this for sure will help

Reply:
Using `u32`s is fine but you should still check the boundaries, because `0u32 - 1 = u32::MAX`.